### PR TITLE
Add support for deployment to GKE

### PR DESCRIPTION
**Pull request by henrybell**
This PR restructures the repository and adds code/config to support deployment of the Autoscaler to Kubernetes (GKE) as well as to Google Cloud Functions, fixing #67. As well as this, #71 and #62 are additionally fixed as a result of changes to the service account provisioning/permissions within the Terraform code.

Because this PR is large and touches a significant proportion of the codebase, thorough testing before merging is recommended -- we can set aside some time for this if helpful.

### CODE REVIEW

**Comment by bgood** [comment]
Thanks for the PR!

I've started working through it, but you are right it is a big one.  One thing we'll need to update are the GitHub Actions to run unit tests. I think the change is simply to update the working directories.

**Comment by henrybell** [comment]
I fixed the `markdownlinter` findings and modified the paths for the unit tests -- will check the results next time the workflows run (currently waiting for approval). Thanks!

**Comment by bgood** [comment]
Did a quick run through of the per-project Cloud Functions deployment.  Looks like the working directory exports in the READMEs need to be updated.  Here is an example from the per-project README, `export AUTOSCALER_DIR="$(pwd)/autoscaler/terraform/per-project"`.

- [ ] terraform/cloud-functions/distributed/README.md
- [ ] terraform/cloud-functions/per-project/README.md

File: README.md
Comment (bgood) [inline README.md:140]: Unused links.

File: terraform/gke/README.md
Comment (bgood) [inline terraform/gke/README.md:20]: Add link back to monitoring section.

**Review by bgood — CHANGES_REQUESTED** [review]
Looking good. Made a few suggestions around cleaning links in the markdown.

File: terraform/gke/README.md
Comment (henrybell) [reply terraform/gke/README.md:238]: I have these docs changes made on another branch, will add to the PR once tested.

**Comment by bgood** [comment]
@davidcueva , this change is looking good.  Given the size and scope would you also be able to do a quick review?

**Comment by henrybell** [comment]
Closing to recreate from new `master` branch following #75.

**Pull request closed without merging**
